content/aspdotnet-controllers-and-routes/reading/controllers-parameters/index.md

```diff
@@ -0,0 +1,174 @@
+---
+title: "Controllers with Parameters"
+date: 2023-02-06T14:09:15-06:00
+draft: false
+weight: 2
+originalAuthor: Sally Steuterman # to be set by page creator
+originalAuthorGitHub: gildedgardenia # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+Now that you know the basics of action methods and controllers, we can start to add some more variables into the mix.
+Some action methods can take in parameters in the form of query strings or sections of the URL path.
+Passing this URL data to the controller is one step closer to more flexible web applications. 
+
+On the previous page, we learned that a route is the mechanism by which a request path gets assigned to a controller within our application.
+In this section, we’ll further explore routes and how data is transferred from a webpage with a given route to a specific controller.
+
+## Query Strings are URL Data
+
+A brief refresher: **query strings** are additional bits of information tacked onto the ends of urls.
+They contain data in key-value pairs.
+
+```console
+   www.galaxyglossary.net?aKey=aValue&anotherKey=anotherValue&thirdKey=thirdValue
+```
+
+## Controllers and Parameters
+
+### Query Strings
+
+We can pass different parameters into an action method.
+Let's add a method called `Welcome()` to `HelloController.cs`. 
+
+```csharp {linenos = table}
+   // GET: /<controller>/welcome
+   public IActionResult Welcome(string name)
+   {
+      return Content("<h1>Welcome to my app, " + name + "!</h1>", "text/html");
+   }
+```
+
+Right now, the `Welcome()` method is using conventional routing, so it will respond to requests at `localhost:5001/hello/welcome`.
+The `Welcome()` method has one argument, `name`. However, when we run the app and navigate to `localhost:5001/hello/welcome`, we can see that `name` has no value.
+
+![Simple webpage that only displays Welcome to my app](pictures/nodefaultvalue.png)
```

Comment:
There is a lot of white space with the photos on this page.  Suggestion of cropping them.

Reply:
Thanks! Some students get confused about whether the footer on the Welcome page is supposed to disappear, so I think the larger images help make it very clear that the page doesn't have one. It is a ton of white space though so I tried re-sizing everything to cut down on the white space and I hope that helps! Let me know your thoughts!

Reply:
Like what you did, looks great!